app/controllers/v0/sign_in_controller.rb

```diff
@@ -175,31 +180,36 @@ def revoke_all_sessions
 
     def logout # rubocop:disable Metrics/MethodLength
       client_id = params[:client_id].presence
-      anti_csrf_token = anti_csrf_token_param.presence
-
-      if client_config(client_id).blank?
+      if client_id && client_config(client_id).blank?
         raise SignIn::Errors::MalformedParamsError.new message: 'Client id is not valid'
       end
 
-      unless load_user(skip_expiration_check: true)
-        raise SignIn::Errors::LogoutAuthorizationError.new message: 'Unable to Authorize User'
+      if (session = SignIn::OAuthSession.find_by(handle: @access_token&.session_handle))
+        credential_type = session.user_verification.credential_type
+        client_id = session.client_id if client_id.blank?
+        anti_csrf_token = anti_csrf_token_param.presence
+
+        SignIn::SessionRevoker.new(access_token: @access_token, anti_csrf_token:).perform
+        delete_cookies if token_cookies
+        logout_redirect = SignIn::LogoutRedirectGenerator.new(credential_type:,
+                                                              client_config: client_config(client_id)).perform
+        sign_in_logger.info('logout', @access_token.to_s)
+        StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_LOGOUT_SUCCESS)
+      else
+        delete_cookies if token_cookies
+        logout_redirect = SignIn::LogoutRedirectGenerator.new(credential_type: nil,
+                                                              client_config: client_config(client_id)).perform
+        sign_in_logger.info('logout error', { errors: 'No valid Session found' })
+        StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_LOGOUT_FAILURE)
       end
 
-      SignIn::SessionRevoker.new(access_token: @access_token, anti_csrf_token:).perform
-      delete_cookies if token_cookies
-
-      sign_in_logger.info('logout', @access_token.to_s)
-      StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_LOGOUT_SUCCESS)
-
-      logout_redirect = SignIn::LogoutRedirectGenerator.new(user: @current_user,
-                                                            client_config: client_config(client_id)).perform
-
       logout_redirect ? redirect_to(logout_redirect) : render(status: :ok)
-    rescue SignIn::Errors::LogoutAuthorizationError, SignIn::Errors::SessionNotAuthorizedError => e
+    rescue SignIn::Errors::SessionNotAuthorizedError => e
       sign_in_logger.info('logout error', { errors: e.message })
       StatsD.increment(SignIn::Constants::Statsd::STATSD_SIS_LOGOUT_FAILURE)
 
-      logout_redirect = SignIn::LogoutRedirectGenerator.new(user: @current_user,
+      credential_type = nil unless defined?(credential_type)
```

Comment:
I don't think this line is doing anything, is it?

Reply:
This was a guard assignment because `credential_type` is only defined partway down in the new method.